frontend/src/components/Analysis/AnalysisDrawer/index.tsx

```diff
@@ -56,75 +57,62 @@ export function AnalysisDrawer({
       featureNameToBucketInfoToPost: SystemsAnalysesBody["feature_to_bucket_info"]
     ) {
       setPageState(PageState.loading);
-      const systemAnalysesReturn: SystemAnalysesReturn | null =
-        await new Promise<SystemAnalysesReturn>((resolve, reject) => {
-          const timeoutID = setTimeout(() => {
-            reject(new Error("timeout"));
-          }, 40000); // 40 seconds
-          backendClient
-            .systemsAnalysesPost({
-              system_ids: activeSystemIDs.join(","),
-              feature_to_bucket_info: featureNameToBucketInfoToPost,
-            })
-            .then(
-              (
-                singleAnalysis:
-                  | SystemAnalysesReturn
-                  | PromiseLike<SystemAnalysesReturn>
-              ) => {
-                clearTimeout(timeoutID);
-                resolve(singleAnalysis);
-              }
-            );
-        }).catch(() => {
-          return null;
-        });
-      if (systemAnalysesReturn === null) {
-        setPageState(PageState.error);
-        return;
-      }
-      const {
-        system_analyses: systemAnalyses,
-        significance_test_info: significanceTestInfos,
-      } = systemAnalysesReturn;
-      /*
-      Take from the first element as the task and type/number of metrics should be 
-      invariant across sytems in pairwise analysis
-      */
-      const firstSystemInfo = activeSystems[0].system_info;
-      const task = firstSystemInfo.task_name;
 
-      const metricToAnalyses = parseFineGrainedResults(
-        task,
-        activeSystems,
-        systemAnalyses
-      );
+      // request times out in 40s
+      const timeoutController = new AbortController();
+      const timeoutID = setTimeout(() => timeoutController.abort(), 0);
```

Comment:
Should it be 40000 instead of 0?

Reply:
Good catch! I forgot to push the latest commit. Fixed.